sway/tree/root.c

```diff
@@ -320,6 +320,23 @@ void root_record_workspace_pid(pid_t pid) {
 	wl_list_insert(&pid_workspaces, &pw->link);
 }
 
+void root_remove_workspace_pid(pid_t pid) {
+	if (!pid_workspaces.prev || !pid_workspaces.next) {
+		return;
+	}
+
+	struct pid_workspace *pw, *tmp;
+	wl_list_for_each_safe(pw, tmp, &pid_workspaces, link) {
+		if (pid == pw->pid) {
+			wl_list_remove(&pw->output_destroy.link);
```

Comment:
It seems like we do this multiple times. Would be nice to extract it into a `pid_workspace_destroy` function.

Reply:
Done